client/gutenberg/extensions/tiled-gallery/view.js

```diff
@@ -1,6 +1,134 @@
 /** @format */
 
+/**
+ * External dependencies
+ */
+import { throttle } from 'lodash';
+import ResizeObserver from 'resize-observer-polyfill';
+
 /**
  * Internal dependencies
  */
+import { RESIZE_RATE_IN_MS, DEFAULT_GALLERY_WIDTH } from './constants';
+import { getActiveStyleName, getLayout } from './layouts';
+
+/**
+ * Styles
+ */
 import './view.scss';
+
+const isBrowser = typeof window !== 'undefined' && typeof document !== 'undefined';
+
+// @TODO this isn't working; need a reliable canonical way to determine editor vs view?
+const isEditor = isBrowser && window.hasOwnProperty( 'wp' ) && window.wp.hasOwnProperty( 'editor' );
+
+const applyNodeSize = ( node, { width, height } ) => {
+	node.style.width = `${ width }px`;
+	node.style.height = `${ height }px`;
+};
+
+/**
+ * Calculate new size for the gallery and apply it
+ */
+const resizeGallery = ( { galleryNode, width, columns, layout } ) => {
+	const tileCount = galleryNode.querySelectorAll( '.tiled-gallery__item' ).length;
+
+	const galleryLayout = getLayout( {
+		columns,
+		layout,
+		tileCount,
+		width,
+	} );
+
+	// eslint-disable-next-line
+	// console.log('|', galleryLayout, options);
+
+	// Resize rows within the gallery
+	galleryNode.childNodes.forEach( ( rowNode, rowIndex ) => {
+		const rowLayout = galleryLayout[ rowIndex ];
+		// eslint-disable-next-line
+		// console.log('| - row:', rowLayout, rowIndex );
+		applyNodeSize( rowNode, rowLayout );
+
+		// Resize tiles within the row
+		const tileNodes = rowNode.querySelectorAll( '.tiled-gallery__item' );
+		tileNodes.forEach( ( tileNode, tileIndex ) => {
+			const tileLayout = rowLayout.tiles[ tileIndex ];
+			// eslint-disable-next-line
+			// console.log('|  ﹂ tile:', tileLayout, tileIndex );
+			applyNodeSize( tileNode, tileLayout );
+		} );
+	} );
+};
+
+/**
+ * Throttled resize event
+ * Compares gallery width to its previous width to decide if to resize it.
+ */
+const throttleOnResize = throttle( entries => {
+	for ( const entry of entries ) {
+		const { width } = entry.contentRect;
+		// Don't resize if width didn't chance
+		if ( width !== entry.target.getAttribute( 'data-width' ) ) {
+			// Store width for later comparison
+			entry.target.setAttribute( 'data-width', width );
+			resizeGallery( {
+				columns: parseInt( entry.target.getAttribute( 'data-columns' ), 10 ),
+				galleryNode: entry.target,
+				layout: getActiveStyleName( entry.target.className ),
+				width,
+			} );
+		}
+	}
+}, RESIZE_RATE_IN_MS );
+
+/**
+ * Get different galleries on the page
+ *
+ * @returns {Array} List of gallery nodes on the page
+ */
+const getGalleries = () => {
+	return document ? [ ...document.querySelectorAll( '.wp-block-jetpack-tiled-gallery' ) ] : [];
```

Comment:
Is this providing safety? Should likely be `'object' === typeof document`. If we keep this, lets use `isBrowser` defined above.

I question the utility of this method, we can likely inline it below and use `for...of`